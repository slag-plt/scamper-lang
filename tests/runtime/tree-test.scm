(struct leaf (value))

(struct node (left right))

(define tree-size
  (lambda (t)
    (if (leaf? t)
        1
        (+ (tree-size (node-left t))
           (tree-size (node-right t))))))

(define tree-to-list
  (lambda (t)
    (if (leaf? t)
        (list (leaf-value t))
        (append (tree-to-list (node-left t))
                (tree-to-list (node-right t))))))

(define t1
  (node (leaf "a")
        (node (leaf "b")
              (leaf "c"))))

t1

(leaf-value (node-left (node-right t1)))

(tree-size t1)

(tree-to-list t1)