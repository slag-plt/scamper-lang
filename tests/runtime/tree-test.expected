(struct node (struct leaf "a") (struct node (struct leaf "b") (struct leaf "c")))
"b"
3
(list "a" "b" "c")
